WordPressCom-Stats-iOS/InsightsTableViewController.m

```diff
@@ -407,6 +904,92 @@ - (void)retrieveStats
              self.sectionData[@(StatsSectionInsightsTodaysStats)] = summary;
          }
      }
+                                              commentsAuthorCompletionHandler:^(StatsGroup *group, NSError *error)
```

Comment:
So many completion handlers :)  A few questions:  
- it looks like many of the handlers share a common pattern. Maybe some of these could be refactored to reuse code? 
- wondering if it would make sense to have a weak self reference here. 
- if `andOverallCompletionHandler` reloads the table view why reload the sections individually?  Maybe because completion happens asynchronously?